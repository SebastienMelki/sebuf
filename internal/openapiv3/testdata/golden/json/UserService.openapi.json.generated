{"components":{"schemas":{"FieldViolation":{"description":"FieldViolation describes a single validation error for a specific field.","properties":{"description":{"description":"Human-readable description of the validation violation (e.g., 'must be a valid email address', 'required field missing')","type":"string"},"field":{"description":"The field path that failed validation (e.g., 'user.email' for nested fields). For header validation, this will be the header name (e.g., 'X-API-Key')","type":"string"}},"required":["field","description"],"type":"object"},"Request":{"description":"Generic request message","properties":{"data":{"description":"Request data","type":"string"},"id":{"description":"Request ID","type":"string"}},"type":"object"},"User":{"description":"User message used by multiple services","properties":{"email":{"description":"User email","type":"string"},"id":{"description":"User ID","type":"string"},"name":{"description":"User name","type":"string"},"role":{"description":"User role","type":"string"}},"type":"object"},"ValidationError":{"description":"ValidationError is returned when request validation fails. It contains a list of field violations describing what went wrong.","properties":{"violations":{"description":"List of validation violations","items":{"$ref":"#/components/schemas/FieldViolation"},"type":"array"}},"required":["violations"],"type":"object"}}},"info":{"title":"UserService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/api/v1/users/create":{"post":{"description":"Create user","operationId":"CreateUser","parameters":[{"description":"User authentication token","in":"header","name":"X-User-Token","required":true,"schema":{"format":"uuid","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"CreateUser","tags":["UserService"]}},"/api/v1/users/get":{"post":{"description":"Get user","operationId":"GetUser","parameters":[{"description":"User authentication token","in":"header","name":"X-User-Token","required":true,"schema":{"format":"uuid","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/Request"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"GetUser","tags":["UserService"]}},"/api/v1/users/update":{"post":{"description":"Update user","operationId":"UpdateUser","parameters":[{"description":"User authentication token","in":"header","name":"X-User-Token","required":true,"schema":{"format":"uuid","type":"string"}}],"requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"description":"Successful response"},"400":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/ValidationError"}}},"description":"Validation error"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"UpdateUser","tags":["UserService"]}}}}