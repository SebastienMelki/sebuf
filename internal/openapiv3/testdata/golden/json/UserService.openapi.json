{"components":{"schemas":{"CreateUserRequest":{"description":"Simple request message","properties":{"email":{"description":"User email","type":"string"},"name":{"description":"User name","type":"string"}},"type":"object"},"DeleteUserRequest":{"description":"Delete user request","properties":{"id":{"description":"User ID to delete","type":"string"}},"type":"object"},"DeleteUserResponse":{"description":"Delete user response","properties":{"message":{"description":"Status message","type":"string"},"success":{"description":"Success indicator","type":"boolean"}},"type":"object"},"GetUserRequest":{"description":"Get user request","properties":{"id":{"description":"User ID to retrieve","type":"string"}},"type":"object"},"UpdateUserRequest":{"description":"Update user request","properties":{"id":{"description":"User ID","type":"string"},"user":{"$ref":"#/components/schemas/User"}},"type":"object"},"User":{"description":"Simple response message","properties":{"email":{"description":"User email","type":"string"},"id":{"description":"User ID","type":"string"},"name":{"description":"User name","type":"string"}},"type":"object"}}},"info":{"title":"UserService API","version":"1.0.0"},"openapi":"3.1.0","paths":{"/api/v1/users":{"post":{"description":"Create a new user","operationId":"CreateUser","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/CreateUserRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"CreateUser","tags":["UserService"]}},"/api/v1/users/delete":{"post":{"description":"Delete user","operationId":"DeleteUser","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/DeleteUserRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/DeleteUserResponse"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"DeleteUser","tags":["UserService"]}},"/api/v1/users/get":{"post":{"description":"Get user by ID","operationId":"GetUser","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/GetUserRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"GetUser","tags":["UserService"]}},"/api/v1/users/update":{"post":{"description":"Update user","operationId":"UpdateUser","requestBody":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/UpdateUserRequest"}}},"required":true},"responses":{"200":{"content":{"application/json":{"schema":{"$ref":"#/components/schemas/User"}}},"description":"Successful response"},"default":{"content":{"application/json":{"schema":{"properties":{"code":{"type":"integer"},"error":{"type":"string"}},"type":"object"}}},"description":"Error response"}},"summary":"UpdateUser","tags":["UserService"]}}}}